### PR TITLE
Adding software autonight detection

### PR DESCRIPTION
**Pull request by chisht**
Adding software logic for night detection. Its under option -S, default is disabled.

### CODE REVIEW

File: autonight/autonight.c [hunk 1]
```diff
@@ -165,7 +178,209 @@ int main(int argc, char *argv[]) {
 
         return 0;
 }
+#define ISP_INFO  "/proc/jz/isp/isp_info"
+#define EXPOSURE  "ISP exposure log2 id: "
+#define IRIDIX    "ISP Iridix strength : "
```
Comment (jmtatsch) [inline autonight/autonight.c]: Just curious, what does that iridix strenght actually mean?

**Comment by nik0** [comment]
Feel free to add the binary if it doesn’t exists...

**Comment by jmtatsch** [comment]
Yes add an autonight_app folder, just like motor_app

File: autonight/autonight.c (hunk 1 above)
Comment (chisht) [reply autonight/autonight.c]: BTW https://github.com/Dafang-Hacks/drivers/blob/16770a7c7f67520b85f8120adf9e0035357e7c91/isp/tx-isp/apical-isp/tx-isp-core.c#L2258 is the only source I can find to explain iridix and the rest of the parameters.

**Comment by ortegafernando** [comment]
@jmtatsch I think you could also close this as you has already merged [https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/pull/739](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/pull/739)

**Pull request merged**